resources/build/trigger-definitions.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+#
+# This file maps specific paths to build triggers
+#
+
+# the base folder for each pattern does not need to be included, nor oem folders
+# e.g. android='common/predictive-text|common/lexical-model-types'
+# will expand into android='^(android|(oem/[^/]+/android)|common/predictive-text|common/lexical-model-types)'
+
+watch_android='web|common/predictive-text|common/lexical-model-types'
+watch_ios='web|common/predictive-text|common/lexical-model-types'
+watch_linux='common/engine'
+watch_mac='common/engine'
+watch_web='common/predictive-text|common/lexical-model-types'
+
+# Windows currently builds Developer and Desktop, so we need everything from common,developer,web
+watch_windows='common|developer|web'
+
+#
+# Available build configurations and VCS identifeirs; identifiers are somewhat inconsistent due 
+# to legacy updates and changes.
+#
+
+# Test Build Configurations
+
+# One day we may have tests that run on all platform changes
+bc_test_all=()
+
+bc_test_android=(KeymanAndroid_TestPullRequests KeymanAndroid_TestSamplesAndTestProjects)
+bc_test_ios=(Keyman_iOS_TestPullRequests Keyman_iOS_TestSamplesAndTestProjects)
+bc_test_linux=(KeymanLinux_TestPullRequests Keyman_Common_KPAPI_TestPullRequests_Linux)
+bc_test_mac=(Keyman_KeymanMac_PullRequests Keyman_Common_KPAPI_TestPullRequests_macOS)
+bc_test_windows=(KeymanDesktop_TestPullRequests KeymanDesktop_TestPrRenderOnScreenKeyboards Keyman_Common_KPAPI_TestPullRequests_Windows)
+bc_test_web=(Keymanweb_TestPullRequests Keyman_Common_LMLayer_TestPullRequests)
+# Keymanweb_TestPullRequestRegressions : currently this is timing out so disabled until we have
```

Comment:
This sounds like a significant (though acceptable?) drawback to using a script to trigger configurations - is there no way for the script to note that a configuration is "disabled" on TC and not trigger?

It's not optimal, but maybe we could add a common config parameter `TC_ENABLED=true` to all configs and condition a first step for each on that?

```
if [ $TC_ENABLED = false ]; then
  # Terminate the CI run with a clear "SKIPPED" message
fi
```

Of course, if there's a simple enough way for a script to detect the actual enablement state of its configuration, drawing directly from that would likely be best.

Reply:
You can't actually 'pause' a configuration in TC, only pause its triggers. Given this is the trigger script, it's the only place we can do it 😢.  This is not ideal but easier than adding variables etc.

Cancelling a build works but is weird and marks it as failed in the GH status check.

What's more, TeamCity+GitHub only allow "success" and "failure" outcomes for checks so it actually becomes confusing -- TC's status publisher would say either "TeamCity build passed" or "TeamCity build failed" -- with no option for nuance. A real shame. So much so that I experimented with a custom status publisher but it's not worth the extra hoops.

I'm not thrilled by the solution here but at least it's clear and easy enough for us to re-enable.

Reply:
Hooray for incoming trigger-PR noise.  😢  Didn't know that aspect of the specifics, so... yeah, makes sense.

Reply:
Yep. TeamCity does so much but always seems to implement it just slightly ... wonkily! 😁

Reply:
> What's more, TeamCity+GitHub only allow "success" and "failure" outcomes for checks so it actually becomes confusing -- TC's status publisher would say either "TeamCity build passed" or "TeamCity build failed" -- with no option for nuance. 

Isn't this a limitation of the GitHub API that allows only `error`, `failure` (which both show up as failed in GitHub), `pending`, or `success`? (https://developer.github.com/v3/repos/statuses/)

Reply:
Yes - the TC side of the limitation is that you can't set the status message. I would set it to "success" + "build skipped" -- but TC can't do that itself.